modules/tinymce/src/plugins/link/test/ts/browser/SelectedTextLinkTest.ts

```diff
@@ -11,17 +12,40 @@ UnitTest.asynctest('browser.tinymce.plugins.link.SelectedTextTest', (success, fa
   LinkPlugin();
   SilverTheme();
 
-  TinyLoader.setupLight(function (editor, onSuccess, onFailure) {
+  TinyLoader.setupLight((editor, onSuccess, onFailure) => {
     const tinyApis = TinyApis(editor);
     const doc = TinyDom.fromDom(document);
 
-    Pipeline.async({},
-      Log.steps('TBA', 'Link: complex selections should preserve the text', [
-        TestLinkUi.sClearHistory,
+    const sTextToDisplayShown = UiFinder.sExists(Body.body(), '.tox-label:contains("Text to display")');
+    const sTextToDisplayHidden = UiFinder.sNotExists(Body.body(), '.tox-label:contains("Text to display")');
+
+    const sOpenDialog = (textToDisplayVisible: boolean = true) => GeneralSteps.sequence([
+      tinyApis.sExecCommand('mcelink'),
+      UiFinder.sWaitForVisible('wait for link dialog', TinyDom.fromDom(document.body), '[role="dialog"]'),
+      textToDisplayVisible ? sTextToDisplayShown : sTextToDisplayHidden
+    ]);
+
+    Pipeline.async({}, [
+      TestLinkUi.sClearHistory,
+      Log.stepsAsStep('TINY-5205', 'Link: basic text selection with existing link should preserve the text when changing URL', [
+        tinyApis.sSetContent('<p><a href="http://oldlink/">word</p>'),
```

Comment:
Should there be a closing `</a>` tag?

Reply:
Yep there should be, though it won't technically matter. I'll fix it up either way though.

Reply:
Fixed